### PR TITLE
Refactor `Block`

### LINKED ISSUE #10907: replace block fields with BlockBody
### Describe the feature

currently Block has a flattened blockbody as its fields:

https://github.com/paradigmxyz/reth/blob/97cc9dd8ecaaf7ebfa0b401ed080da28dc791376/crates/primitives/src/block.rs#L40-L47

https://github.com/paradigmxyz/reth/blob/97cc9dd8ecaaf7ebfa0b401ed080da28dc791376/crates/primitives/src/block.rs#L557-L566

if we want to abstract a block as 

```rust
trait Block {type Body;}
```

then we need to use the container type

see also:

https://github.com/paradigmxyz/reth/blob/97cc9dd8ecaaf7ebfa0b401ed080da28dc791376/crates/primitives/src/block.rs#L301-L306

## TODO
* replace fields with `BlockBody`

### Additional context

fyi @refcell 

https://github.com/alloy-rs/alloy/issues/1290

### PR DESCRIPTION
**Pull request by roynalnaruto**
Closes #10907

### CODE REVIEW

File: crates/primitives/src/block.rs [hunk 1]
```diff
@@ -125,74 +103,17 @@ impl Block {
...
-    }
-
-    /// Returns only the blob transactions, if any, from the block body.
-    #[inline]
-    pub fn blob_transactions(&self) -> Vec<&TransactionSigned> {
-        self.blob_transactions_iter().collect()
-    }
-
-    /// Returns an iterator over all blob versioned hashes from the block body.
-    #[inline]
```
Comment (mattsse) [inline crates/primitives/src/block.rs:153]: none of those should be removed I think

File: crates/primitives/src/block.rs [hunk 2]
```diff
@@ -234,32 +155,31 @@ impl BlockWithSenders {
         (self.block, self.senders)
     }
 
-    /// Returns an iterator over all transactions in the block.
-    #[inline]
-    pub fn transactions(&self) -> impl Iterator<Item = &TransactionSigned> + '_ {
-        self.block.body.iter()
-    }
```
Comment (mattsse) [inline crates/primitives/src/block.rs:241]: why remove?

**Review by mattsse — COMMENTED** [review]
this is looking good, what's missing?

File: crates/primitives/src/block.rs (hunk 1 above)
Comment (roynalnaruto) [reply crates/primitives/src/block.rs:153]: I moved them to `BlockBody`, but I'll keep them and call the `BlockBody`'s methods.

File: crates/primitives/src/block.rs (hunk 2 above)
Comment (roynalnaruto) [reply crates/primitives/src/block.rs:241]: similarly, will reinstate.

File: crates/primitives/src/block.rs [hunk 3]
```diff
@@ -128,52 +103,102 @@ impl Block {
...
+    pub fn size(&self) -> usize {
+        self.header.size() + self.body.size()
     }
+}
 
-    /// Returns whether or not the block contains any EIP-7702 transactions.
-    #[inline]
-    pub fn has_eip7702_transactions(&self) -> bool {
-        self.body.iter().any(|tx| tx.is_eip7702())
+mod block_rlp {
```
Comment (mattsse) [inline crates/primitives/src/block.rs:113]: could we add some docs here why we need to do this manually? (rlp flatten)

**Review by mattsse — APPROVED** [review]
lgtm, only doc nit

**Comment by mattsse** [comment]
tysm @roynalnaruto 
@klkvr got this over the line because this pr is a conflict magnet, and this now unblocks a few additional features

**Comment by roynalnaruto** [comment]
Thank you @klkvr

**Pull request merged**